newshomepages/utils.py

```diff
@@ -463,3 +470,165 @@ def intcomma(value):
+        ".newspack-lightbox",  # Common takover ad
+        ".pbs_loc",  # PBS site location picker
+        "#onetrust-consent-sdk",
+        ".gdpr-huffpost-cookiewall",
+        "#header-cts",
+        ".pum-overlay",  # Common takeover ad
+        "#pico_prompt",
+        "#pico_header",
+        ".mol-ads-cmp",  # Footer toaster thing on some sites
+        ".gdpr-glm-standard",  # GDPR blockers on some sites
+        "#didomi-host",  # Common cookies popup
+        ".fc-ab-root",  # Ad blocker popup
+        ".fancybox-overlay",  # Popup overlay
+        ".fancybox-overlay-fixed",
+        ".ab_widget_container_popin-image",  # Popover
+        ".ab_widget_container_popin-image_content",
+        ".ab_widget_container_popin-image_overlay",
+        ".fc-ab-root",  # Popover
+        ".modal-backdrop",  # Modal popup
+        "#dth-borderless-modal-alt",
+        "#floatingBlockerGuide",  # Popup overlay
+        ".ctct-popup-overlay",  # Popup overlay
+        ".ctct-popup-wrapper",
+        ".message-container",  # Popup
+        ".teg-container",
+        'div[id^="sp_message_container"]',  # Popover
+        ".wisepops-root",  # Popover
+        "#wisepops-root",
+        ".widget_eu_cookie_law_widget",  # GDPR poup
+        ".fancybox-overlay",  # Ad overlay
+        ".signup-box",  # Popup ad
+        ".newspack-lightbox",  # Popup
+        ".adunit-googleadmanager",  # Google ad
+        ".mc-modal",  # Newsletter popup
+        ".mc-modal-bg",
+        ".blockNavigation",
+        ".ab-iam-root",
+        ".omaha-background",  # Takeover ad
+        ".sqs-popup-overlay",  # Overlay popup
+        "#gcomPromo",  # Overlay found on Brazilian sites
+        ".Campaign",
+    ]
+    target_str = ",".join(target_list)
+    javascript = f"""
+        document.querySelectorAll('{target_str}').forEach(el => el.remove());
+        var styleSheet = document.createElement('style');
+        styleSheet.innerText = '{target_str} {{ display: none !important; }}';
+        document.head.appendChild(styleSheet);
+     """
+    return javascript
+
+
+def _load_new_page_disable_javascript(
+    context: BrowserContext, url: str, wait_seconds: int = 5, handle: str = None
+):
+    """Load the page with javascript blocking."""
+    # Create an empty tab
+    page = context.new_page()
+    # Open the page
+    print(f"Opening {url}")
+    page.goto(url, timeout=60000)
+
+    # Give it a beat
+    print(f"Waiting {wait_seconds} seconds")
+    time.sleep(wait_seconds)
+    common_javascript = _get_common_blocking_javascript()
+
+    print("Executing common JavaScript")
+    page.evaluate(common_javascript)
+
+    # If there's custom javascript for this site, run it
+    custom_javascript = get_javascript(handle)
+    if custom_javascript:
+        print("Executing custom JavaScript")
+        try:
+            page.evaluate(custom_javascript)
+        except Exception as e:
+            raise click.ClickException(str(e))
```

Comment:
Since we have abstracted this outside of the click environment, how about just letting the exception raise on its own?